File: dotnet/xmscommon/AMQUndeliveredException.cs
using System;
using System.Text;

namespace OpenAMQ
{
    /// <summary>
    /// Thrown when a message has been bounced by the broker, indicating it could not be delivered.
    /// </summary>
    public class AMQUndeliveredException : AMQException
    {
        private object _bounced;
        
        public AMQUndeliveredException(int errorCode, string message, object bounced)
            : base(errorCode, message)
        {
            _bounced = bounced;
        }
    }
}

File: java/src/org/openamq/AMQUndeliveredException.java
package org.openamq;

/**
 * Generic AMQ exception.
 */
public class AMQUndeliveredException extends AMQException
{
    private Object _bounced;

    public AMQUndeliveredException(int errorCode, String msg, Object bounced)
    {
        super(errorCode, msg);

        _bounced = bounced;
    }

    public Object getUndeliveredMessage()
    {
        return _bounced;
    }

}


